libs/vkd3d/command.c

```diff
@@ -408,6 +380,7 @@ static void vkd3d_wait_for_gpu_timeline_semaphores(struct vkd3d_fence_worker *wo
             if (FAILED(hr = d3d12_fence_signal(current->fence, counter_value)))
                 ERR("Failed to signal D3D12 fence, hr %#x.\n", hr);
 
+            d3d12_fence_dec_ref(current->fence);
```

Comment:
Should dec_ref *after* the InterlockedDecrement below perhaps?

Reply:
Should we still fix this given that the entire function gets reworked in the next commit anyway?

Reply:
Fixed, but this gets reworked in the following commits anyway.